examples/language-modeling/run_clm.py

```diff
@@ -390,17 +391,19 @@ def main():
             if model_args.torch_dtype in ["auto", None]
             else getattr(torch, model_args.torch_dtype)
         )
-        model = AutoModelForCausalLM.from_pretrained(
-            model_args.model_name_or_path,
-            from_tf=bool(".ckpt" in model_args.model_name_or_path),
-            config=config,
-            cache_dir=model_args.cache_dir,
-            revision=model_args.model_revision,
-            use_auth_token=True if model_args.use_auth_token else None,
-            torch_dtype=torch_dtype,
-        )
+        with lazy_load_for_parallelism(tensor_parallel_size=training_args.tensor_parallel_size):
```

Comment:
Instead of using this context before any `AutoModelForCausalLM` when you want to use TP, invoke this context, why didn't you use a neuron/TP specialized child class of `AutoModelForCausalLM` and override the methods that need to be modified ?